File: src/engine/ChangeLens.Core/Comparisons/Models/ComparisonTargetKind.cs
namespace ChangeLens.Core.Comparisons.Models;

/// <summary>
///     Defines the supported kinds of comparison target.
/// </summary>
public enum ComparisonTargetKind
{
    /// <summary>
    ///     A local branch reference.
    /// </summary>
    Local,

    /// <summary>
    ///     A cached remote-tracking branch reference.
    /// </summary>
    RemoteTracking,
}

File: src/desktop/src-tauri/src/comparisons/models/comparison_target_kind.rs
use serde::{Deserialize, Serialize};

/// Defines the source of a comparison target.
#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
#[serde(rename_all = "camelCase")]
pub enum ComparisonTargetKind {
    /// A local branch target.
    Local,
    /// A cached remote-tracking branch target.
    RemoteTracking,
}
